indico/web/forms/fields.py

```diff
@@ -255,7 +255,8 @@ class CategoryField(HiddenField):
     widget = JinjaWidget('forms/category_picker_widget.html')
 
     def __init__(self, *args, **kwargs):
-        self.select_leaf_only = kwargs.pop('select_leaf_only', False)
+        self.allow_action_on_cat_with_events = kwargs.pop('allow_action_on_cat_with_events', True)
```

Comment:
What about `allow_events` and `allow_subcats`? The current names are awfully long and if the short ones are not verbose enough - that's what docstrings are for..

```python
    """Field that lets you select a category.

    :param allow_events: Whether to allow selecting a category that
                         contains events.
    :param allow_subcats: Whether to allow selecting a category that
                          contains subcategories.
    """
```